wallet/wallet.go

```diff
@@ -2373,6 +2373,175 @@ func (w *Wallet) GetTransactions(startBlock, endBlock *BlockIdentifier,
+	// confirm during the time we receive it from the backend and check it
+	// in the database.
+	// We therefore store the best block height and will use it for quering the
+	// database.
+	_, bestHeight, err := chainClient.GetBestBlock()
+	if err != nil {
+		return nil, false, errors.New("cannot get best block")
+	}
+
+	// Get the transaction information from directly calling the backend
+	// endpoint.
+	var txResult *btcjson.TxRawResult
+	switch client := chainClient.(type) {
+	case *chain.RPCClient:
+		txResult, err = client.GetRawTransactionVerbose(txHash)
+		if err != nil {
+			return nil, false, err
+		}
+	case *chain.BitcoindClient:
+		txResult, err = client.GetRawTransactionVerbose(txHash)
+		if err != nil {
+			return nil, false, err
+		}
+	case *chain.NeutrinoClient:
+		return nil, false, errors.New("not supported with neutrino client")
+	}
+
+	// Set the block hash from the provided string and get the block height
+	// from it.
+	//
+	// TODO: Fetching block heights by their hashes is inherently racy
+	// because not all block headers are saved but when they are for SPV the
+	// db can be queried directly without this.
+	var blockHash *chainhash.Hash
+	var height int32 = -1
+	if txResult.BlockHash != "" {
+		blockHashBytes, err := hex.DecodeString(txResult.BlockHash)
+		if err != nil {
+			return nil, false, err
+		}
+
+		// The byte is reversed due to different endianness in response
+		// and call.
+		reversed := make([]byte, len(blockHashBytes))
+		for i, n := range blockHashBytes {
+			j := len(blockHashBytes) - i - 1
+			reversed[j] = n
+		}
+		blockHash, err = chainhash.NewHash(reversed)
+		if err != nil {
+			return nil, false, err
+		}
+
+		// Obtain the block height from the backend.
+		switch client := chainClient.(type) {
+		case *chain.RPCClient:
+			header, err := client.GetBlockHeaderVerbose(blockHash)
+			if err != nil {
+				return nil, false, err
+			}
+			height = header.Height
+		case *chain.BitcoindClient:
+			height, err = client.GetBlockHeight(blockHash)
+			if err != nil {
+				return nil, false, err
+			}
+		case *chain.NeutrinoClient:
+			height, err = client.GetBlockHeight(blockHash)
+			if err != nil {
+				return nil, false, err
+			}
+		}
+
+		// We know that the transaction definitively is in a specific
+		// block height and set both start and end height to it.
+		bestHeight = height
+	}
+
+	// Populate with additional data if this transaction exists in the
+	// database.
+	var summary *TransactionSummary
+	var inDB bool
+	err = walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		rangeFn := func(details []wtxmgr.TxDetails) (bool, error) {
+			// TODO: probably should make RangeTransactions not
+			// reuse the details backing array memory.
+			dets := make([]wtxmgr.TxDetails, len(details))
+			copy(dets, details)
+			details = dets
+
+			for i := range details {
+				dbTx := makeTxSummary(dbtx, w, &details[i])
```

Comment:
Hmm, could this just be:

```
for i, txDetail := range details {
        txDetail := txDetail
        dbTx := makeTxSummary(dbtx, w, &txDetail)
```

to avoid copying the whole slice?

Reply:
I was a bit confused about this TODO comment. Note though, that I used the code from the `GetTransactions` method above here. But I have made your code suggestion in the squashed commit.